### ISSUE TITLE
[REVIEW]: eSCAPE: parallel global-scale landscape evolution model

### ISSUE THREAD

**Issue by whedon** [openjournals/joss-reviews#964]
**Submitting author:** @tristan-salles (<a href="http://orcid.org/0000-0001-6095-7689">Tristan Salles</a>)
**Repository:** <a href="https://github.com/Geodels/eSCAPE" target ="_blank">https://github.com/Geodels/eSCAPE</a>
**Version:** v1.0.0
**Editor:** @kthyng
**Reviewer:** @sgrieve, @kbarnhart 
**Archive:** <a href="https://doi.org/10.5281/zenodo.1419844" target="_blank">10.5281/zenodo.1419844</a>

## Status

[![status](http://joss.theoj.org/papers/56aa3dfe0cc458fcef10e998a945ba7e/status.svg)](http://joss.theoj.org/papers/56aa3dfe0cc458fcef10e998a945ba7e)

Status badge code:

```
HTML: <a href="http://joss.theoj.org/papers/56aa3dfe0cc458fcef10e998a945ba7e"><img src="http://joss.theoj.org/papers/56aa3dfe0cc458fcef10e998a945ba7e/status.svg"></a>
Markdown: [![status](http://joss.theoj.org/papers/56aa3dfe0cc458fcef10e998a945ba7e/status.svg)](http://joss.theoj.org/papers/56aa3dfe0cc458fcef10e998a945ba7e)
```
**Reviewers and authors**:

Please avoid lengthy details of difficulties in the review thread. Instead, please create a new issue in the <a href="https://github.com/Geodels/eSCAPE/tree/master/joss" target="_blank">target repository</a> and link to those issues (especially acceptance-blockers) in the review thread below. (For completists: if the target issue tracker is also on GitHub, linking the review thread in the issue or vice versa will create corresponding breadcrumb trails in the link target.)

## Reviewer instructions & questions

@sgrieve & @kbarnhart, please carry out your review in this issue by updating the checklist below. If you cannot edit the checklist please:
1. Make sure you're logged in to your GitHub account
2. Be sure to accept the invite at this URL: https://github.com/openjournals/joss-reviews/invitations

The reviewer guidelines are available here: https://joss.theoj.org/about#reviewer_guidelines. Any questions/concerns please let @kthyng know.

✨ **Please try and complete your review in the next two weeks** ✨ 

### Conflict of interest

- [x] As the reviewer I confirm that I have read the [JOSS conflict of interest policy](https://github.com/openjournals/joss/blob/master/COI.md) and that there are no conflicts of interest for me to review this work.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/Geodels/eSCAPE/tree/master/joss">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Version:** Does the release version given match the GitHub release (v1.0.0)?
- [x] **Authorship:** Has the submitting author (@tristan-salles) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the function of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Authors:** Does the `paper.md` file include a list of authors with their affiliations?
- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **References:** Do all archival references that should have a DOI list one (e.g., papers, datasets, software)?

### Conflict of interest

- [x] As the reviewer I confirm that I have read the [JOSS conflict of interest policy](https://github.com/openjournals/joss/blob/master/COI.md) and that there are no conflicts of interest for me to review this work.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/Geodels/eSCAPE/tree/master/joss">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Version:** Does the release version given match the GitHub release (v1.0.0)?
- [x] **Authorship:** Has the submitting author (@tristan-salles) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?

### Functionality

- [ ] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [ ] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [ ] **Automated tests:** Are there automated tests or manual steps described so that the function of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Authors:** Does the `paper.md` file include a list of authors with their affiliations?
- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **References:** Do all archival references that should have a DOI list one (e.g., papers, datasets, software)?

**Comment by whedon** [comment]
Hello human, I'm @whedon, a robot that can help you with some common editorial tasks. @sgrieve it looks like you're currently assigned as the reviewer for this paper :tada:.

**:star: Important :star:**

If you haven't already, you should seriously consider unsubscribing from GitHub notifications for this (https://github.com/openjournals/joss-reviews) repository. As a reviewer, you're probably currently watching this repository which means for GitHub's default behaviour you will receive notifications (emails) for all reviews 😿

To fix this do the following two things:

1. Set yourself as 'Not watching' https://github.com/openjournals/joss-reviews:

![watching](https://cloud.githubusercontent.com/assets/4483/20250593/64d7ce48-a9de-11e6-9225-d3dfb3e48e68.png)

2. You may also like to change your default settings for this watching repositories in your GitHub profile here: https://github.com/settings/notifications

![notifications](https://cloud.githubusercontent.com/assets/4483/20250604/79eb3144-a9de-11e6-9e51-0fe940966689.png)


For a list of things I can do to help you, just type:

```
@whedon commands
```

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by whedon** [comment]
PDF failed to compile for issue #964 with the following error: 

 /app/vendor/ruby-2.3.4/lib/ruby/2.3.0/find.rb:43:in `block in find': No such file or directory (Errno::ENOENT)
	from /app/vendor/ruby-2.3.4/lib/ruby/2.3.0/find.rb:43:in `collect!'
	from /app/vendor/ruby-2.3.4/lib/ruby/2.3.0/find.rb:43:in `find'
	from /app/vendor/bundle/ruby/2.3.0/bundler/gems/whedon-2caaf5cba519/lib/whedon/processor.rb:57:in `find_paper_paths'
	from /app/vendor/bundle/ruby/2.3.0/bundler/gems/whedon-2caaf5cba519/bin/whedon:32:in `prepare'
	from /app/vendor/bundle/ruby/2.3.0/gems/thor-0.20.0/lib/thor/command.rb:27:in `run'
	from /app/vendor/bundle/ruby/2.3.0/gems/thor-0.20.0/lib/thor/invocation.rb:126:in `invoke_command'
	from /app/vendor/bundle/ruby/2.3.0/gems/thor-0.20.0/lib/thor.rb:387:in `dispatch'
	from /app/vendor/bundle/ruby/2.3.0/gems/thor-0.20.0/lib/thor/base.rb:466:in `start'
	from /app/vendor/bundle/ruby/2.3.0/bundler/gems/whedon-2caaf5cba519/bin/whedon:99:in `<top (required)>'
	from /app/vendor/bundle/ruby/2.3.0/bin/whedon:22:in `load'
	from /app/vendor/bundle/ruby/2.3.0/bin/whedon:22:in `<main>'

**Comment by kthyng** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by whedon** [comment]
PDF failed to compile for issue #964 with the following error: 

 /app/vendor/ruby-2.3.4/lib/ruby/2.3.0/find.rb:43:in `block in find': No such file or directory (Errno::ENOENT)
	from /app/vendor/ruby-2.3.4/lib/ruby/2.3.0/find.rb:43:in `collect!'
	from /app/vendor/ruby-2.3.4/lib/ruby/2.3.0/find.rb:43:in `find'
	from /app/vendor/bundle/ruby/2.3.0/bundler/gems/whedon-2caaf5cba519/lib/whedon/processor.rb:57:in `find_paper_paths'
	from /app/vendor/bundle/ruby/2.3.0/bundler/gems/whedon-2caaf5cba519/bin/whedon:32:in `prepare'
	from /app/vendor/bundle/ruby/2.3.0/gems/thor-0.20.0/lib/thor/command.rb:27:in `run'
	from /app/vendor/bundle/ruby/2.3.0/gems/thor-0.20.0/lib/thor/invocation.rb:126:in `invoke_command'
	from /app/vendor/bundle/ruby/2.3.0/gems/thor-0.20.0/lib/thor.rb:387:in `dispatch'
	from /app/vendor/bundle/ruby/2.3.0/gems/thor-0.20.0/lib/thor/base.rb:466:in `start'
	from /app/vendor/bundle/ruby/2.3.0/bundler/gems/whedon-2caaf5cba519/bin/whedon:99:in `<top (required)>'
	from /app/vendor/bundle/ruby/2.3.0/bin/whedon:22:in `load'
	from /app/vendor/bundle/ruby/2.3.0/bin/whedon:22:in `<main>'

**Comment by kthyng** [comment]
ok @sgrieve and @kbarnhart I have this ready for you now. Let me know if you have any questions.

@arfon Do you have any guidance for how I can determine why the pdf is not compiling?

**Comment by arfon** [comment]
@whedon generate pdf

**Comment by kthyng** [comment]
@arfon You must have the magic touch.

**Comment by arfon** [comment]
@kthyng - sometimes @whedon just needs a little encouragement :-)

**Comment by kbarnhart** [comment]
Since I've just loaded on a bunch of issues about documentation, testing, and installation, I thought I'd also make a note here that I think the core functionality of eSCAPE is great, the tutorials are very nicely done, and the docker image works as expected.

**Comment by sgrieve** [comment]
I would like to thank the author for this contribution, it has been a very enjoyable experience to explore this software and I am pleased to recommend publication, pending the minor issues highlighted below being resolved.

# Version
I noticed that there is a version 1.0.1 on GitHub, but the submission corresponds to version 1.0.0. It would be good to clarify this, so that the current stable version is the one that is reviewed.

# Installation
In the first instance I explored the code using Docker. Everything worked out of the box, and running the test notebook yielded the correct results with no surprises or problems. I would however, suggest that a little more detail in the main `readme.md` file on how to use Kitematic and Docker may be useful, as I imagine that this is the most likely method of install for new users, and those not familiar with Docker may get held up on this step.

One issue I noticed was that there is no `requirements.txt` file at the base of the repo, containing all of the python dependencies and their version requirements as listed on the HPC install page on the wiki, this would save people a bit of hassle during the HPC install process.

# Functionality

Working my way through the extensive tutorials and examples showed me the range of applications for eSCAPE, and I am happy that the functional claims made in the documentation and the paper are valid. More broadly I believe that eSCAPE will be a valuable tool for researchers across the geosciences, and that it is providing a novel solution to the common problem of scale in the modeling of surface processes.

# Performance

I have been unable to run this software on an HPC system (due to my current lack of access to one) so have only been able to explore the performance of eSCAPE on my MacBook. Where performance claims have been made, they have been in line with my experience of running eSCAPE. As I have tested larger and larger model runs, I have been pleasantly surprised by the compute times and believe that the performance of this model is in line with other similar model's performance, if not exceeding them.

# Documentation

Going through the documentation I am impressed at the detail and quality of what has been produced. Most of the issues I would have flagged have already been resolved following the issues created by @kbarnhart.

One further issue I noticed was that in the homepage of the API documentation, there is a table listing submodules (`mesher`, `flow`, `pit`, `tools`). This table has no associated information about what these submodules are/what they do. Similarly, on the contents pane of the API docs there are links to these submodules (`eSCAPE.mesher`, `eSCAPE.flow`, `eSCAPE.pit`, `eSCAPE.tools`), but these links do not contain any information. I realize that Sphinx can be a pain for stuff like this, but if possible it would be good to either populate these pages if there is relevant information to go into them, or remove the links to avoid confusion.

Some other minor issues:

- The `About` section of the wiki just contains some images, and no information.
- The `Contact` section of the wiki links back to the wiki homepage, and provides no contact information
- In the `runModel` notebook for the Australia tutorial, the ParaView command `Warp by Scalar` is called `Wrap by Scalar`. (I think this issue might be repeated in some of the other notebooks.)

# Tests

The test notebook that is provided works well to ensure that the code is running as expected, and provides detailed information to users to begin debugging if things do not go as expected. In an ideal world there would be some kind of CI set up to run automated tests, but I appreciate the difficulties in doing this. I know that some UK research councils have a national Jenkins instance which supports automated testing and I don't know if that is something you can explore with your funders?

# Paper

The paper clearly outlines the motivation for the development of eSCAPE, and the key theoretical underpinnings of the work. It reads clearly and follows the general pattern for JOSS papers.

Overall I am delighted to have had the opportunity to review this paper, and have been thoroughly impressed by the overall quality of the software, documentation and tutorials. I hope to continue using the software in the future! :rocket:

**Comment by kbarnhart** [comment]
I thought I'd give a quick update on the progress of my review. As noted by @sgrieve I've made most of my comments in the form of issues. @tristan-salles has dealt with most of these impressively fast. This has been very helpful as I was able to attempt HPC installation based on the revised HPC instructions instead of the original ones.

I am testing the HPC installation instructions so between the two reviewers we will have tested all installation options. As of yesterday evening am hung up on a step related to the `petsc4py` installation done by my institutions Research Computing staff. @tristan-salles and I have been iterating on improvements to the HPC installation instructions on [eSCAPE Issue #3](https://github.com/Geodels/eSCAPE/issues/3). I don't expect that resolving my HPC issues will take too long. Once I've been able to confirm HPC install I'll be able finalize my review. 

I anticipate that I will recommend acceptance with minor revisions (many of which @tristan-salles has already completed).

**Comment by tristan-salles** [comment]
A massive thank you to @kbarnhart and @sgrieve for their quick and super valuable inputs, comments & feedbacks!

### Comments from @sgrieve

I think I have pretty much addressed all the comments. In details I have:
+ fixed the version to 1.0.0 which is now consistent with the one from the paper
+ added a `requirements.txt` file
+ updated the wiki with a step by step guide in regards to installing and using Docker and Kitematic ([see](https://github.com/Geodels/eSCAPE/wiki/Using-Docker))
+ updated the API to populated the submodule table and removing unnecessary submodules links ([see](https://escape-api.github.io/index.html)) 
+ updated the [About section of the wiki](https://github.com/Geodels/eSCAPE/wiki/About)
+ updated the [Contact section of the wiki](https://github.com/Geodels/eSCAPE/wiki/Contact)
+ fixed the typo in the demo python notebooks (`warp by scalar` to `wrap by scalar`)

Regarding the testing issue, I have added a `test installation` example in the `eSCAPE-demo` repository that provides two comparisons related to:
+ the _expected values_ that you should obtained if your installation is successful
+ runtime for both _serial_ and _parallel_ simulations
This test should take less than 1 minute. An example on how to visualise **eSCAPE** outputs in [**Paraview**](https://www.paraview.org/download/) is also provided.

### Comments from @kbarnhart 

I think I have answered most of the issues and comments there (see the history of closed issues above). There is still a remaining open one: **Improve eSCAPE HPCC installation instructions Geodels/eSCAPE#3**.

**Comment by sgrieve** [comment]
Thanks for responding to my comments so quickly @tristan-salles!

@kthyng I'm satisfied that all of my comments have been addressed, so pending the resolution of the HPC install issues that @kbarnhart has raised I'm happy to formally accept!

**Comment by kthyng** [comment]
I see that @sgrieve is ready to accept, and I see the issue referenced above that is in progress on @kbarnhart's end. Thanks for the great progress and work, @tristan-salles, @sgrieve, and @kbarnhart!

**Comment by kbarnhart** [comment]
Alright. I'm completing my review now @kthyng .

## Barnhart eSCAPE review

I'm happy to recommend acceptance contingent on minor revision.

This is an excellent contribution and I look forward to having applications that
require using it. 

My primary concern is with HPC installation instructions. As I expect that many
major research-grade applications (and/or coupling with additional models) will 
require a user to compile the code, I think that ironing these instructions out 
is important. 

## Explanation for few checkboxes I didn't check.

I don't think any of these things should prevent this paper from moving forward. 

I didn't check '**Installation**: Does installation proceed as outlined in the 
documentation?' as I was not able to surmount some of my HPC platform issues and
thus I never successfully installed this on an HPC. Ultimately the issue seems 
to be related to the details of the Fortran compiler used in the backend of the 
Intel compiler version that the staff at the University of Colorado Research 
computing used to build PETSc. 
@tristan-salles is in no way at fault in this and was very helpful throughout my attempt to compile. Based on my attempts I think
that the installation instructions are **almost** sufficient (see comments 
below).

I also didn't check '**Installation instructions**: Is there a clearly-stated 
list of dependencies? Ideally these should be handled with an automated package 
management solution.' I think that both the `requirements.txt` file and the HPC 
installation instructions need to be revised (more details below).

I didn't check '**Automated tests**: Are there automated tests or manual steps 
described so that the function of the software can be verified?' because there 
are no automated tests. However there are tests that meet the [JOSS "good" 
requirements](https://joss.readthedocs.io/en/latest/review_criteria.html#tests). I think there are reasonable reasons that there are no automated 
tests (e.g. need a platform with PETSc, etc, to run the tests).

## Recommended revision to to installation instructions. 

I think that there are two items that relate to installation that need to be
addressed. The first relates to the `requirements.txt` and the second relates
to the HPC install instructions wiki page. 

## requirements.txt

I think that the `requirements.txt` file needs to be further revised in order
to include the minimum version numbers (where applicable). I also think that
having the `fillit` and `meshplex` package in the `requirements.txt` file is a 
bit awkward as these are not packages that are distributed on standard package
distributors (conda, pip). Further the `meshplex` package is a modified version
forked by @tristan-salles. I see no problem with these as dependencies, but 
putting these in a `requirements.txt` file means that a command like. 

```
pip install -r requirements.txt
```

will fail because these packages are not findable. 

## HPC install instructions wiki

I think that the [HPC installation page on the wiki](https://github.com/Geodels/eSCAPE/wiki/Installation-on-HPC)
needs to be further revised to not be so specific to one HPC platform. I also 
think that text related to the fortran compiler used is confusing.

### Specificity regarding platform 

Sections like [this one on example dependencies](https://github.com/Geodels/eSCAPE/wiki/Installation-on-HPC#dependencies-installation-example)
show loading specific modules like petsc built with gcc and open MPI. I would 
recommend removing these lines and replacing with regular (not code block) text 
that says something like:

> Before you begin compiling eSCAPE and dependencies, make sure you have set up 
your compute environment to include python, OpenMPI, a supported compiler, and 
PETSc. 

Now that a `requirements .txt` file exists, all the pip install commands can be
replaced by a single call 

```
pip install -r requirements.txt
```

Combining all the code needed to install on the Artemis HPC at USyD is useful, 
but perhaps should go on another page called "Example installation scripts used on specific HPC platforms"
Once I get my platform working I would happily submit my script there. This would help
draw the distinction for a user that there are *general* requirements, and that there are
*specific* examples that worked on a particular platform.

### Fortran compiler

There are two places in the examples where the fortran compiler F90 is assigned 
to something else (ifort in one place and gfortran in another). I will admit to 
not knowing much about different fortran compilers (or different compilers in general...) 
and thus these instructions are confusing to me. I might guess that I should be 
using gfortran if I used gcc instead of the intel compiler, and ifort if I used 
the intel compiler. I would recommend only writing 

```
export F90=something
```

in one place on the HPC instructions and providing a bit of context as to why
someone should use gfortran or ifort in this situation. 

### Minor comments about the example
The [example PBS script](https://github.com/Geodels/eSCAPE/wiki/Installation-on-HPC#example-of-pbs-script)
assumes that a user has downloaded the `eSCAPE-demo` repository and dealt with all
of its requirements. It should be revised to include information about downloading
the demo repository. I also think that it would be more appropriate on another wiki
page.

**Comment by kbarnhart** [comment]
@kthyng a quick note that I've now closed all the issues I opened in the eSCAPE repository.

**Comment by kthyng** [comment]
@kbarnhart Has @tristan-salles completed or otherwise addressed the minor revisions you noted or are there any outstanding issues?

**Comment by kbarnhart** [comment]
Yes! Absolutely! No outstanding issues remain. I am happy to recommend
publication. Sorry if that wasn’t previously clear.

**Comment by kthyng** [comment]
Ok thanks @kbarnhart, just wanted to be sure.

@arfon This paper has been accepted by the reviewers (@kbarnhart and @sgrieve)!!!! Ready for publication!!

**Comment by kthyng** [comment]
@tristan-salles Please create an archive (on Zenodo, figshare, or other) and post the archive DOI in this issue.

**Comment by tristan-salles** [comment]
Hi @kthyng here is the zenodo archive:
[![DOI](https://zenodo.org/badge/DOI/10.5281/zenodo.1419844.svg)](https://doi.org/10.5281/zenodo.1419844)

DOI: 10.5281/zenodo.1419844

**Comment by kthyng** [comment]
@whedon set 10.5281/zenodo.1419844 as archive

**Comment by whedon** [comment]
OK. <a href="https://doi.org/10.5281/zenodo.1419844" target="_blank">10.5281/zenodo.1419844</a> is the archive.

**Comment by kthyng** [comment]
Thanks @tristan-salles!

**Comment by arfon** [comment]
@sgrieve, @kbarnhart - many thanks for your reviews here and to @kthyng for editing this submission ✨ 

@tristan-salles - your paper is now accepted into JOSS and your DOI is  https://doi.org/10.21105/joss.00964 :zap: :rocket: :boom:

**Comment by whedon** [comment]
:tada::tada::tada: Congratulations on your paper acceptance! :tada::tada::tada:

If you would like to include a link to your paper from your README use the following code snippets:

```
Markdown:
[![DOI](http://joss.theoj.org/papers/10.21105/joss.00964/status.svg)](https://doi.org/10.21105/joss.00964)

HTML:
<a style="border-width:0" href="https://doi.org/10.21105/joss.00964">
  <img src="http://joss.theoj.org/papers/10.21105/joss.00964/status.svg" alt="DOI badge" >
</a>

reStructuredText:
.. image:: http://joss.theoj.org/papers/10.21105/joss.00964/status.svg
   :target: https://doi.org/10.21105/joss.00964
```
This is how it will look in your documentation:

[![DOI](http://joss.theoj.org/papers/10.21105/joss.00964/status.svg)](https://doi.org/10.21105/joss.00964)

**We need your help!**

Journal of Open Source Software is a community-run journal and relies upon volunteer effort. If you'd like to support us please consider doing either one (or both) of the the following:

- Volunteering to review for us sometime in the future. You can add your name to the reviewer list here: http://joss.theoj.org/reviewer-signup.html
- Making a small donation to support our running costs here: https://www.flipcause.com/secure/cause_pdetails/Mjk3Nzk=
